file/file_prefetch_buffer.h

```diff
@@ -448,6 +450,17 @@ class FilePrefetchBuffer {
     return false;
   }
 
+  // Performs tuning to calculate readahead_size.
+  void ReadAheadSizeTuning(uint64_t offset, size_t n) {
+    UpdateReadAheadSizeForUpperBound(offset, n);
+
+    if (readaheadsize_cb_ != nullptr && readahead_size_ > 0) {
+      size_t updated_readahead_size = 0;
+      readaheadsize_cb_(offset, readahead_size_, updated_readahead_size);
+      readahead_size_ = updated_readahead_size;
```

Comment:
This will be for current prefetching. Next will be update_readahead_size*2;

Reply:
Ohh! So we want to keep the original_readahead_size *2 for the next prefetch instead of updated_readahead_size*2

Reply:
> Ohh! So we want to keep the original_readahead_size *2 for the next prefetch instead of updated_readahead_size * 2 in order to not reduce the readahead size drastically for next prefetch.

Addressed the changes